records_mover/db/postgres/postgres_copy_options.py

```diff
@@ -0,0 +1,268 @@
+    #
+    #  Specifies the quoting character to be used when a data value is
+    #  quoted. The default is double-quote. This must be a single
+    #  one-byte character. This option is allowed only when using CSV
+    #  format.
+    #
+
+    postgres_options['quote'] = hints['quotechar']
+    quiet_remove(unhandled_hints, 'quotechar')
+
+    # ESCAPE
+    #
+    #  Specifies the character that should appear before a data
+    #  character that matches the QUOTE value. The default is the same
+    #  as the QUOTE value (so that the quoting character is doubled if
+    #  it appears in the data). This must be a single one-byte
+    #  character. This option is allowed only when using CSV format.
+    #
+
+    if not hints['doublequote']:
+        cant_handle_hint(fail_if_cant_handle_hint, 'doublequote', hints)
+    else:
+        quiet_remove(unhandled_hints, 'doublequote')
+
+    if hints['escape'] is not None:
+        cant_handle_hint(fail_if_cant_handle_hint, 'escape', hints)
+    else:
+        quiet_remove(unhandled_hints, 'escape')
+
+    # FORCE_QUOTE
+    #
+    #  Forces quoting to be used for all non-NULL values in each
+    #  specified column. NULL output is never quoted. If * is
+    #  specified, non-NULL values will be quoted in all columns. This
+    #  option is allowed only in COPY TO, and only when using CSV
+    #  format.
+    #
+
+    if hints['quoting'] != 'minimal':
+        cant_handle_hint(fail_if_cant_handle_hint, 'quoting', hints)
+    else:
+        quiet_remove(unhandled_hints, 'quoting')
+
+    # FORCE_NOT_NULL
+    #
+    #  Do not match the specified columns' values against the null
+    #  string. In the default case where the null string is empty,
+    #  this means that empty values will be read as zero-length
+    #  strings rather than nulls, even when they are not quoted. This
+    #  option is allowed only in COPY FROM, and only when using CSV
+    #  format.
+    #
+    # ENCODING
+    #
+    #  Specifies that the file is encoded in the encoding_name. If
+    #  this option is omitted, the current client encoding is
+    #  used. See the Notes below for more details.
+
+    # TODO: Test different encodings
+    postgres_options['encoding'] = hints['encoding']
+    quiet_remove(unhandled_hints, 'encoding')
+
+    # TODO: Get this to work on read and write and test what we
+    # produce vs what we can accept
+    quiet_remove(unhandled_hints, 'record-terminator')
+
+    # TODO: Get this to work on read and write and test what we
+    # produce vs what we can accept
+    quiet_remove(unhandled_hints, 'datetimeformattz')
+
+    # TODO: Get this to work on read and write and test what we
+    # produce vs what we can accept
+    quiet_remove(unhandled_hints, 'datetimeformat')
+
+    # TODO: Get this to work on read and write and test what we
+    # produce vs what we can accept
+    quiet_remove(unhandled_hints, 'timeonlyformat')
+
+    # TODO: Get this to work on read and write and test what we
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker